### PR TITLE
Shawn/capture version

### PR DESCRIPTION
**Pull request by ShawnDhanani**

### CODE REVIEW

File: StockCount/View/OptionViewController.swift [hunk 1]
```diff
@@ -19,8 +20,16 @@ class OptionViewController : UIViewController {
         super.viewDidLoad()
         
         let versionString = (Bundle.main.infoDictionary?["CFBundleVersion"] as? String) ?? ""
-        lblVersion.text = "version".localized + versionString
-        
+        lblVersion.text = "version".localized + versionString + "\nCapture ver"
+        CaptureHelper.sharedInstance.getVersionWithCompletionHandler { (result, version) in
+            DispatchQueue.main.async {
```
[the remark this answers is not in the document]
Comment (EricGlaenzer) [reply StockCount/View/OptionViewController.swift]: I think we would like to display 2 versions. The version of the Application and the version of Capture. They should be separate.

[the remark this answers is not in the document]
Comment (EricGlaenzer) [reply StockCount/View/OptionViewController.swift]: I would recommend to display a default string for the Capture version indicating that it's being fetched, and maybe an error string if it couldn't be fetched.

File: StockCount/View/OptionViewController.swift [hunk 2]
```diff
@@ -19,16 +19,15 @@ class OptionViewController : UIViewController {
...
+        let appVer = (Bundle.main.infoDictionary?["CFBundleVersion"] as? String) ?? ""
+        updateVersion(appVer, "fetching".localized)
         CaptureHelper.sharedInstance.getVersionWithCompletionHandler { (result, version) in
-            DispatchQueue.main.async {
-                let majorVersionString = String(version?.major ?? 0)
-                let middleVersionString = String(version?.middle ?? 0)
-                let minorVersionString = String(version?.minor ?? 0)
-                let captureVersion = majorVersionString + "." + middleVersionString + "." + minorVersionString
-                self.lblVersion.text = "version".localized + versionString + "\nCapture \(captureVersion)"
+            var captureVersion = "error".localized
```
Comment (EricGlaenzer) [inline StockCount/View/OptionViewController.swift]: we should check if result is successful. If it is then version will be correct.

**Comment by EricGlaenzer** [comment]
Please answer the previous discussion and resolve them when you push the new code. Thanks!

**Comment by ShawnDhanani** [comment]
Updated code for above discussions.
Please have a review again.

File: StockCount/View/OptionViewController.swift (hunk 2 above)
Comment (EricGlaenzer) [reply StockCount/View/OptionViewController.swift]: I don't see the check for result success?

File: StockCount/View/OptionViewController.swift [hunk 3]
```diff
@@ -24,7 +24,7 @@ class OptionViewController : UIViewController {
         updateVersion(appVer, "fetching".localized)
         CaptureHelper.sharedInstance.getVersionWithCompletionHandler { (result, version) in
             var captureVersion = "error".localized
-            if version != nil {
+            if version != nil && result == SKTResult.E_NOERROR {
```
Comment (ShawnDhanani) [inline StockCount/View/OptionViewController.swift]: Added result check code on last commit.
https://github.com/SocketMobile/stockcount-ios/pull/35/commits/8c59739b1b2668f2d58b0b2cce9f715e1aed9958

**Comment by ShawnDhanani** [comment]
https://github.com/SocketMobile/stockcount-ios/pull/35/commits/8c59739b1b2668f2d58b0b2cce9f715e1aed9958

Please check this commit.

File: StockCount/View/OptionViewController.swift (hunk 2 above)
Comment (ShawnDhanani) [reply StockCount/View/OptionViewController.swift]: updated code on last commit.

https://github.com/SocketMobile/stockcount-ios/pull/35/commits/8c59739b1b2668f2d58b0b2cce9f715e1aed9958

**Pull request merged**